server/channels/app/post.go

```diff
@@ -1338,6 +1350,10 @@ func (a *App) DeletePost(c request.CTX, postID, deleteByID string) (*model.Post,
 		a.deleteFlaggedPosts(post.Id)
 	})
 
+	if err = a.RemoveNotifications(c, post, channel, team); err != nil {
```

Comment:
If `RemoveNotifications` is the only one using the team, and it also has access to the `App` object, why let the Team get fetched at this level instead of inside the `RemoveNotifications` function? Trying to fail early if the team is wrongly set?

Reply:
Makes sense. Fixed now.